pipeline/coordinator.go

```diff
@@ -188,6 +199,46 @@ func NewStubCoordinatorOpts(strategy Strategy, statusClient clients.TranscodeSta
 // This has the main logic regarding the pipeline strategy. It starts jobs and
 // handles processing the response and triggering a fallback if appropriate.
 func (c *Coordinator) StartUploadJob(p UploadJobPayload) {
+	// A bit hacky - this is effectively a dummy job object to allow us to reuse the runHandlerAsync and
```

Comment:
I'm not sure I fully understand this here - mind elaborating why it's hacky? Also, if the InputCopyToS3 fails for some reason, does it get reflected in the async handler correctly? Maybe we should trigger this condition in a test to see what happens?

Reply:
The tests I added to coordinator_test.go cover some error scenarios e.g. https://github.com/livepeer/catalyst-api/pull/463/files#diff-d93900b04efde86efa555721e97460e1caf9b798bbdae4a68a1f4105c0500160R227
I'll check if there are any other cases that could be added